File: src/main/kotlin/by/mksn/kwitapi/service/impl/UserServiceImpl.java
package by.mksn.kwitapi.service.impl;

import by.mksn.kwitapi.entity.User;
import by.mksn.kwitapi.repository.UserRepository;
import by.mksn.kwitapi.service.UserService;

public class UserServiceImpl implements UserService {

    private final UserRepository userRepository;

    public UserServiceImpl(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public User findByUsername(String username) {
        return userRepository.findByUsername(username);
    }
}

File: src/main/kotlin/by/mksn/kwitapi/service/impl/UserServiceImpl.kt
package by.mksn.kwitapi.service.impl

import by.mksn.kwitapi.controller.RegistrationDetails
import by.mksn.kwitapi.model.User
import by.mksn.kwitapi.model.UserRepository
import by.mksn.kwitapi.service.AbstractCrudService
import by.mksn.kwitapi.service.UserService
import by.mksn.kwitapi.service.tryCallJPA

class UserServiceImpl(val userRepository: UserRepository) : AbstractCrudService<User, Long>(userRepository), UserService {

    override fun register(registrationDetails: RegistrationDetails) {
        TODO("not implemented")
    }

    override fun findByEmail(username: String): User?
            = tryCallJPA { userRepository.findByEmail(username) }
}
